### PR TITLE
Add build functionality via grunt

### LINKED ISSUE #81: Create a build script
We need to create a build script for everything needed to deploying on GAE, currently some stuff is needed to be done manually.

* Vulcanize index.html
* Make sure components/webrtc-adapter exists due to test pages in manual/ depend on it (testrtc no longer requires access to components/ after vulcanize (it pulls all (almost) files together into one file)).

After this is done we can setup jenkins #65.

### PR DESCRIPTION
**Pull request by KaptenJansson**
* Add back package.json so it's more convenient setting up the repo using "npm install"
* Change bower components folder to the components/ in the repo root folder rather than outside the repo, this was done to keep the same folder structure for both the vulcanized and non vulcanized version.
* Add grunt build steps that copies all necessary files to deploy and vulcanizes index.html in a out/ folder.
* The previous bullet and the one before enables developing on an non vulcanized index.html version in src/ (start GAE using the app.yaml in the root) and after running "grunt build" you can run the vulcanized version locally in out/ (start GAE using the app.yaml in out/).
* Fixes #81.

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -6,6 +6,42 @@ module.exports = function(grunt) {
...
+        cwd: '.',
+        files: [
+          {src: [
+            'cron.yml',
+            'app.yaml',
+            'src/manual/**',
+            'components/webrtc-adapter/adapter.js'
+            ],
+            dest: 'out',
+            nonull:true,
```
Comment (andresusanopinto) [inline Gruntfile.js]: nit: space.

File: Gruntfile.js [hunk 2]
```diff
@@ -65,7 +105,14 @@ module.exports = function(grunt) {
...
+  grunt.loadNpmTasks('grunt-vulcanize');
+  grunt.loadNpmTasks('grunt-contrib-copy');
+  grunt.loadNpmTasks('grunt-contrib-clean');
 
-  // set default tasks to run when grunt is called without parameters
+  // Set default tasks to run when grunt is called without parameters
   grunt.registerTask('default', ['csslint', 'htmlhint', 'jscs', 'jshint']);
+
+  // Cleans out/ folder, copies files in place and vulcanizes index.html to out/.
+  grunt.registerTask('build', ['clean', 'copy', 'vulcanize']);
```
Comment (andresusanopinto) [inline Gruntfile.js]: Does travis runs this step too? I would like travis to fail if somehow the vulcanize or other steps fail there.

**Comment by andresusanopinto** [comment]
lgtm

File: Gruntfile.js (hunk 2 above)
Comment (KaptenJansson) [inline Gruntfile.js]: yes, grunt build is in .travis.yml and will fail the build if the steps fail.

**Pull request merged**